### PR TITLE
Add the ability to read the VARC condition list

### PR DESCRIPTION
**Pull request by rsheeter**
I love a headache from hand editing ttx in the evening

### CODE REVIEW

File: read-fonts/src/tables/varc.rs [hunk 1]
```diff
@@ -1,17 +1,139 @@
...
     }
+
+    #[test]
+    fn identify_all_conditional_types() {
+        let font = FontRef::new(font_test_data::varc::CONDITIONALS).unwrap();
+        let table = font.varc().unwrap();
+
+        // We should have all 5 condition types in order
+        assert_eq!(
+            (1..=5).collect::<Vec<_>>(),
```
Comment (cmyr) [inline read-fonts/src/tables/varc.rs:38]: iirc you can compare slices/arrays with vecs (there are `PartialEq` impls) so I think you can write, `assert_eq!([1, 2, 3, 4, 5], blah.collect::<Vec<_>>())`

File: resources/codegen_inputs/varc.rs [hunk 2]
```diff
@@ -5,9 +5,32 @@
...
+
+}
+
+table ConditionList {
+    condition_count: u32,
+    #[count($condition_count)]
+    condition_offsets: [Offset32<Condition>],
+}
+
+table AxisIndicesList {
```
Comment (cmyr) [inline resources/codegen_inputs/varc.rs:30]: maybe some docs explaining why these empty tables exist?

File: resources/codegen_inputs/layout.rs [hunk 3]
```diff
@@ -611,14 +611,17 @@ table ConditionSet {
...
+    condition_offsets: [Offset32<Condition>],
 }
 
-///// [Condition Table](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#condition-table)
-//Condition {
-    ///// FIXME: make an enum
-    //no_field: fake,
-//}
+/// [Condition Table](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#condition-table)
+format u16 Condition {
```
Comment (cmyr) [inline resources/codegen_inputs/layout.rs:618]: I would expand on the docs a bit to mention that we are including items that are not yet part of the spec, and then link to whatever the best current reference is?

File: read-fonts/src/tables/varc.rs (hunk 1 above)
Comment (rsheeter) [reply read-fonts/src/tables/varc.rs:38]: I'm getting ```can't compare `[{integer}; 5]` with `Vec<i32>``` so I will leave it for the time being

File: resources/codegen_inputs/varc.rs (hunk 2 above)
Comment (rsheeter) [reply resources/codegen_inputs/varc.rs:30]: Added a terse explanation; I'm not planning to leave them blank very long (subject to meeting load)

File: resources/codegen_inputs/layout.rs (hunk 3 above)
Comment (rsheeter) [reply resources/codegen_inputs/layout.rs:618]: Good idea, done

File: read-fonts/src/tables/varc.rs [hunk 4]
```diff
@@ -1,17 +1,139 @@
...
 
 include!("../../generated/generated_varc.rs");
 
 #[cfg(test)]
 mod tests {
     use crate::{FontRef, TableProvider};
 
+    use super::{Condition, Varc};
+
+    fn read_conditions<'a>(table: &'a Varc) -> Vec<Condition<'a>> {
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply read-fonts/src/tables/varc.rs:13]: Good point, I think I underutilize this. The code then reads more clearly (LTR) which is nice. So updated.

File: read-fonts/src/tables/varc.rs (hunk 1 above)
Comment (cmyr) [reply read-fonts/src/tables/varc.rs:38]: ah okay the issue there might be that you need to be explicit about the integer type, like `[1i32, 2, 3, 4, 5]`

**Pull request merged**